app/src/main/java/ai/elimu/appstore/receiver/InstallCompleteReceiver.java

```diff
@@ -12,7 +12,15 @@
 
     @Override
     public void onReceive(Context context, Intent intent) {
-        installCompleteCallback.onInstallComplete(intent.getData().toString());
+        if ((installCompleteCallback == null) || (intent.getData() == null)) {
+            return;
+        }
+        if (Intent.ACTION_PACKAGE_ADDED.equals(intent.getAction())) {
+            installCompleteCallback.onInstallComplete(intent.getData().getSchemeSpecificPart());
+        } else {
+            installCompleteCallback.onUninstallComplete(intent.getData().getSchemeSpecificPart());
```

Comment:
Why is `onUninstallComplete` used here, if this is an InstallCompleteReceiver?

Reply:
I've refactor the receiver's name.